Img2vec node

cerashdan opened the pull request:
Initial skeleton for Img2Vec node 
Updated docker compose to include Img2Vec node
Updated readme and added the architecture image

cerashdan committed:
Initial commit'

cerashdan committed:
Initial commit'

cerashdan committed:
initial commit

cerashdan committed:
initial commit

cerashdan committed:
adding source for architecture diagrtam

cerashdan committed:
adding source for architecture diagrtam

acmiyaguchi commented on README.md:
```diff
@@ -1,8 +1,13 @@
+# Reinforcement Learning
 
-This is exploration into reinforcement learning with manipulators.
+This is exploration into reinforcement learning with manipulators. We're trying to build a modular environment to train RL-Based agents by providing a standard REST Based Interface to access multiple simulations environment based of Handy Maniupulators.
```
One sentence per line. Also make sure to install the pre-commit hooks: `pip install pre-commit; pre-commit install`. It should include a spell checker.

acmiyaguchi commented on README.md:
```diff
@@ -27,6 +32,9 @@ make build
 make up
+
+# You can enter a running container by running the following command
+docker exec -it [docker_id] /bin/bash
```
Prefer `docker compose exec {container_name} bash`, since you can just refer to the container by name instead of by id.

acmiyaguchi commented on docker-compose.yml:
```diff
@@ -59,3 +59,11 @@ services:
+# image embeddings to another topic
+  img2vec:
+    <<: *base-gpu
+    command: roslaunch --wait finalarm_moveit_config moveit_rviz.launch
```
This probably can just start running without waiting for moveit.

acmiyaguchi commented on ros/img2vec/CMakeLists.txt:
```diff
@@ -0,0 +1,206 @@
+cmake_minimum_required(VERSION 3.0.2)
+project(img2vec)
+
+## Compile as C++11, supported in ROS Kinetic and newer
```
Please remove all the comments that you don't need from this file and from the package.xml. It makes it easier to read.

acmiyaguchi commented on ros/img2vec/scripts/img2vec.py:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
https://github.com/Autobots-Visman/pick-and-place/tree/main/ros/autobots_calibration

Refer to this module if you want a module that reads from a camera and publishes things to a topic. It includes unit testing.

acmiyaguchi replied:
This is actually a much better example, since it reads an image and generates grasp keypoint candidates: https://github.com/ivalab/GraspKpNet/tree/main/ros/gknet_perception

cerashdan committed:
adding imgpublisher as test dependencies

cerashdan committed:
adding imgpublisher as test dependencies

acmiyaguchi committed:
Add initial package for fastapi

Does not work because of network mode inside of the docker environment,
will need to do more debugging to figure out issues.

acmiyaguchi committed:
Use pip-compile instead of poetry for ros dependencies

acmiyaguchi committed:
Update state controller with functioning state and index page

acmiyaguchi committed:
Use htmx for displaying the main ui

acmiyaguchi committed:
Add notes on using tailscale as a workaround for wsl2 host

acmiyaguchi committed:
Merge pull request #6 from Autobots-Visman/fastapi

Add initial state controller for interface to reinforcement learning agent

acmiyaguchi committed:
Move agent code into a separate directory

acmiyaguchi committed:
fix: remove poetry reference from ros package

acmiyaguchi committed:
Merge pull request #8 from Autobots-Visman/reorg-agent

Reorganize agent code into separate directory

cerashdan committed:
adding image publisher logic

cerashdan committed:
resolving merge

cerashdan committed:
receving image succesfully

cerashdan committed:
updated dependencies

cerashdan committed:
adding img2vec image embedding with pytorch
